### PR TITLE
Add Server Side Encryption

### PR DESCRIPTION
**Pull request by sethpollack**
## What Changed & Why

Added an optional param `serverSideEncryption`.
## PR Checklist
- [x ] Add tests - just passed them.
- [x ] Add documentation

### CODE REVIEW

**Comment by LevelbossMike** [comment]
I just restarted the Travis build and this LGTM. ~~I'll merge this as soon as the build passes~~ (one small nitpick change requested). Just out of curiosity what's the reason behind using server side encryption for index.html files? Company policy or real safety reasons?

**Comment by LevelbossMike** [comment]
@sethpollack as the documentation states that there are only two supported values for the serverside-encryption key and we are asserting on passing `undefined` as a value (which seems to be unsupported as a value for the encryption key) I would actually  be more comfortable as a safety measure to see the tests be changed and check if the param is filtered out when the user did not specify the serverSideEncryption option.

Did you check that the plugin still functions as expected without this option present for users that don't need this? (which it should do according to the code changes just asking)

File: lib/s3.js [hunk 1]
```diff
@@ -55,6 +56,10 @@ module.exports = CoreObject.extend({
       CacheControl: cacheControl
     };
 
+    if (serverSideEncryption) {
```
Comment (LevelbossMike) [inline lib/s3.js:59]: can we please add a test for this conditional

**Review by LevelbossMike — CHANGES_REQUESTED** [review]

**Comment by sethpollack** [comment]
Just a company policy. Do these changes help?

**Comment by LevelbossMike** [comment]
@sethpollack tests are failing. I suggest we add two tests to the s3 client to make sure we are only adding this when the serverSideEncryption option is passed (one for not passing the option, one for passing). And the same to index-nodetest.

**Comment by sethpollack** [comment]
Oh oops. Git added it back to the options. This should fix it. 

I'll add the tests now.

**Comment by sethpollack** [comment]
Actually, I need to add this to the activate hook too. 

Will push up the changes soon.

**Review by LevelbossMike — APPROVED** [review]

**Comment by LevelbossMike** [comment]
:tada: thank you so much for this contribution @sethpollack

**Pull request merged**